src/builtins.ts

```diff
@@ -771,6 +772,62 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      compiler.currentType = compiler.program.stringInstance.type;
+
+      if (checkTypeOptional(typeArguments, reportNode, compiler)
+        | checkArgsOptional(operands, 0, 1, reportNode, compiler)) {
```

Comment:
I guess it should be `checkTypeOptional(...)` || checkArgsOptional(...)`.

Reply:
This check is going away because I'm going to use `evaluateConstantType() `

Reply:
This should be all set :)

Reply:
BTW @MaxGraey this kind of check already exists in the compiler elsewhere which is why I did this in the first place. Is there a chance this particular check is not good in the other places either?

Reply:
A bitwise or is used to make both sides execute so all the errors become reported. Iirc I introduced that to replace otherwise [unnecessary longish code](https://github.com/AssemblyScript/assemblyscript/commit/252b843c4bdb0947be70a682b5ba38e1878dcd5c#diff-2c8ebd9fe94e61d77bc89956497d80daL2066).

Reply:
Thanks for clarifying. I'm glad I was able to resolve this problem with a function that already exists.